roles/zabbix_agent/tasks/psk_secret.yml

```diff
@@ -12,17 +12,21 @@
     - name: AutoPSK | Check for existing TLS PSK file | Windows
       ansible.windows.win_stat:
         path: "{{ zabbix_agent_tlspskfile }}"
-      register: zabbix_agent_tlspskcheck
+      register: zabbix_agent_tlspskcheck_windows
```

Comment:
I don't see how `zabbix_agent_tlspskcheck` could be set twice since the two choices for the `ansible_os_family` variable are mutually exclusive.

Reply:
In the original code, there is two tasks that use the "register" keyword for the same variable. This two task are, indeed, mutual exclusive. But Ansible also set the variable if the task is skipped.

> If a task fails or is skipped, Ansible still registers a variable with a failure or skipped status, unless the task is skipped based on tags.

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html